plugins/sqliteendorsements/main.go

```diff
@@ -0,0 +1,250 @@
+// Copyright 2021 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"database/sql"
+	"fmt"
+
+	"reflect"
+
+	"github.com/hashicorp/go-plugin"
+	_ "github.com/mattn/go-sqlite3"
+
+	"veraison/common"
+)
+
+type SqliteEndorsementStore struct {
+	common.BaseEndorsementStore
+	db   *sql.DB
+	path string
+}
+
+func (e *SqliteEndorsementStore) GetName() string {
+	return "sqlite"
+}
+
+// Init opens the database connection.
+// Expected parameters:
+//    dbPath -- the path to the database file.
+func (e *SqliteEndorsementStore) Init(args common.EndorsementStoreParams) error {
+	dbPath, found := args["dbPath"]
+	if !found {
+		return fmt.Errorf("dbPath not specified inside FetcherParams")
+	}
+
+	dbConfig := fmt.Sprintf("file:%s?cache=shared", dbPath)
+	db, err := sql.Open("sqlite3", dbConfig)
+	if err != nil {
+		return err
+	}
+
+	e.db = db
+	e.path = dbPath
+	e.Queries = map[string]common.Query{
+		"hardware_id":         e.GetHardwareId,
+		"software_components": e.GetSoftwareComponents,
+	}
+
+	return nil
+}
+
+// Close the database connection.
+func (e *SqliteEndorsementStore) Close() error {
+	return e.db.Close()
+}
+
+func (e *SqliteEndorsementStore) GetHardwareId(args common.QueryArgs) (common.QueryResult, error) {
+	var platformId string
+	var result []interface{}
+
+	platformIdArg, ok := args["platform_id"]
+	if !ok {
+		return nil, fmt.Errorf("Missing mandatory query argument 'platform_id'")
+	}
+
+	switch platformIdArg.(type) {
+	case string:
+		platformId = platformIdArg.(string)
+	case []interface{}:
+		platformId, ok = platformIdArg.([]interface{})[0].(string)
+		if !ok {
+			return nil, fmt.Errorf("Unexpected type for 'platform_id'; must be a string")
+		}
+	default:
+		t := reflect.TypeOf(platformIdArg)
+		return nil, fmt.Errorf("Unexpected type for 'platform_id'; must be a string; found: %v", t)
+	}
+
+	rows, err := e.db.Query("select hw_id from hardware where platform_id = ?", platformId)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var hwId string
+		if err := rows.Scan(&hwId); err != nil {
+			return nil, err
+		}
+
+		result = append(result, hwId)
+	}
+
+	return result, nil
+}
+
+func (e *SqliteEndorsementStore) GetSoftwareComponents(args common.QueryArgs) (common.QueryResult, error) {
+	//TODO: this method is horrible; do not read this; it needs to be destroyed with fire
```

Comment:
Might be worth adding a polish comment, as this methid improvement is a Work In Progress"

Reply:
I've remove d the comment.